### PR TITLE
Remove the duplicate TransactionIdManager in voltdb

### PR DESCRIPTION
**Pull request by jhugg**
Make all refs point to voltcore version.

Update the javadoc to VoltProcedure.getUniqueId to be long and scary.

Companion:
https://github.com/VoltDB/pro/pull/1416

### CODE REVIEW

File: src/frontend/org/voltdb/iv2/TxnEgo.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
     static final long SEQUENCE_MAX_VALUE = (1L << SEQUENCE_BITS) - 1L;
     static final int PARTITIONID_MAX_VALUE = (1 << PARTITIONID_BITS) - 1;
 
-    // (Copy/Pasted (on purpose) from voltdb.TransactionIdManager)
+    // (Copy/Pasted (on purpose) from voltcore.TransactionIdManager)
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/TxnEgo.java]: You replaced the one in UniqueIdGenerator to call TransactionIdManager.getEpoch(), can this one also be replaced?

**Comment by nshi** [comment]
Review finished.

File: src/frontend/org/voltdb/iv2/TxnEgo.java (hunk 1 above)
Comment (jhugg) [inline src/frontend/org/voltdb/iv2/TxnEgo.java]: I don't know. I mean, yes. I didn't because it says "on purpose". 
I guess I respect comments too much.

**Comment by xinjiacs** [comment]
This looks nice cleanup, we may can clean it as well for TxnEgo as Ning mentioned. But This does not prevent it from merging.

**Comment by xinjiacs** [comment]
I merged master resolving the conflicts, but the github diffs are scary. The new corresponding PR is at: VoltDB/voltdb#4052

**Pull request closed without merging**